applications/ShapeOptimizationApplication/custom_utilities/geometry_utilities.cpp

```diff
@@ -359,4 +382,353 @@ void GeometryUtilities::CalculateUnitNormals()
     }
 }
 
+void GeometryUtilities::CalculateGaussianCurvature() {
+
+    KRATOS_TRY
+
+    // check if surface normals are necessary (only for Taubin method)
```

Comment:
Is it possible to incorporate this loop check into the code inside if statements below. For instance, if technique Taubin and the normalsHasn't computed -> compute Normals and set flag, NormalsComputed True. The same for Meyer and edges.

Reply:
It's a good idea, though I am not sure how to modify the parallel loop then. Because `ExtractEdgeNodes` can't be executed twice or in parallel and `ComputeUnitSurfaceNormals` probably not as well.